src/utilities/modules/vaccines.js

```diff
@@ -304,19 +312,39 @@ export const aggregateLogs = ({
       intervalEndDate
     );
 
-    const timestamp = formatChartDate(
-      new Date(intervalStartDate.getTime() + medianDuration),
-      timestampsByHour
-    );
+    const timeStampUnformated = new Date(intervalStartDate.getTime() + medianDuration);
+    const thisLogsMonth = timeStampUnformated.getMonth();
+    const thisLogsDay = timeStampUnformated.getDay();
+    const shouldDisplayMonth = thisLogsMonth !== currentMonth;
+    const shouldDisplayDays = thisLogsDay !== currentDay;
+
+    const timestamp = formatChartDate(timeStampUnformated, shouldDisplayMonth, shouldDisplayDays);
+
+    currentMonth = thisLogsMonth;
+    currentDay = thisLogsDay;
 
     if (intervalLogs.length === 0) {
       maxLine.push({ temperature: null, timestamp });
       minLine.push({ temperature: null, timestamp });
       return;
     }
 
-    const maxSensorLog = intervalLogs.sorted('temperature', true)[0];
-    const minSensorLog = intervalLogs.sorted('temperature', false)[0];
+    let maxSensorLog = intervalLogs.sorted('temperature', true)[0];
+    let minSensorLog = intervalLogs.sorted('temperature', false)[0];
+
+    if (includeBothEnds) {
+      let shouldUseLast = null;
+      if (intervalStartDate <= startTimestamp && startTimestamp <= intervalEndDate) {
+        shouldUseLast = false;
+      }
+      if (intervalStartDate <= endTimestamp && endTimestamp <= intervalEndDate) {
+        shouldUseLast = true;
+      }
+      if (shouldUseLast !== null) {
+        maxSensorLog = intervalLogs.sorted('timestamp', shouldUseLast)[0];
+        minSensorLog = maxSensorLog;
+      }
```

Comment:
I'm having a hard time following this block. It seems this is just doing checks for the `startDate` and `endDate` parameters and if the intervals are outside of those bounds we do some adjustments (using the same `sensorLog` for max/min). Seems an easier solution is to just reverse max/min when creating the bounds?